cpu/sam0_common/sam0_sdhc/mtd_sdhc.c

```diff
@@ -60,6 +60,36 @@ static int _init(mtd_dev_t *dev)
     return 0;
 }
 
+/*
+ * If SDHC fails to execute a command, re-init usually helps
+ */
+static int _read_blocks_safe(sdhc_state_t *state, uint32_t block, void *dst, uint16_t num)
+{
+    if (sdhc_read_blocks(state, block, dst, num) == 0) {
+        return 0;
+    }
+
+    return sdhc_init(state) || sdhc_read_blocks(state, block, dst, num);
+}
+
+static int _write_blocks_safe(sdhc_state_t *state, uint32_t block, const void *dst, uint16_t num)
+{
+    if (sdhc_write_blocks(state, block, dst, num) == 0) {
+        return 0;
+    }
+
+    return sdhc_init(state) || sdhc_write_blocks(state, block, dst, num);
```

Comment:
I wonder whether the re initialization might lead to a problem with a SD card if the first try of `sdhc_write_blocks` just failed because of `-EBUSY`, for example because the SD card is just still busy with a previous write command.